### PR TITLE
fix(breakout-rooms): enforce rooms order, minor fixes

### LINKED ISSUE #11921: Breakout rooms mixed up in participants editor
Technically, only names could be mixed up, though still confusing. Could happen because of different sorting order of room tokens (2 goes before 1 in alphabetic order).

image

- Sort rooms by display names, not by token
- Don't calculate display names manually, use provided by server:
https://github.com/nextcloud/spreed/blob/09fadadc053f44c0c9834942c41519ceed6f207b/src/components/BreakoutRoomsEditor/BreakoutRoomsParticipantsEditor.vue#L33-L43

_Originally posted by @DorraJaouad in https://github.com/nextcloud/spreed/pull/11649#pullrequestreview-1957078764_

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix #11921
  * sort rooms in getter by id (order of creation, so 1 -> 2 -> 3) _hard to reproduce_
  * use server-provided room names where possible _ParticipantsEditor_
  * ease template in Vue SFC files 
  * add some ?. conditions to avoid TypeErrors _Vue devtools_
  * fix styles to avoid double scrollbar in dialogs _screenshot_
  * don't mount NcDialog for deletion if not called explicitly _console_

TODO:
 - [ ] find another way to restrict image sizes (now it affects user_status)

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
image | image

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖥️ Tested with Desktop client or should not be risky for it 
- [ ] ⛑️ Tests are included or not possible

### CODE REVIEW

File: src/components/BreakoutRoomsEditor/BreakoutRoomsParticipantsEditor.vue [hunk 1]
```diff
@@ -294,8 +295,8 @@ export default {
 		},
 
 		roomName(index) {
-			const roomNumber = index + 1
-			return t('spreed', 'Room {roomNumber}', { roomNumber })
+			return this.breakoutRooms[index]?.displayName
+				?? t('spreed', 'Room manual {roomNumber}', { roomNumber: index + 1 })
```
Comment (nickvergessen) [inline src/components/BreakoutRoomsEditor/BreakoutRoomsParticipantsEditor.vue:299]:
```suggestion
				?? t('spreed', 'Room {roomNumber}', { roomNumber: index + 1 })
```

:confused:

Comment (Antreesy) [reply src/components/BreakoutRoomsEditor/BreakoutRoomsParticipantsEditor.vue:299]: Bah, missed it :see_no_evil: 
Though should help with debugging)

File: src/components/BreakoutRoomsEditor/BreakoutRoomsEditor.vue [hunk 2]
```diff
@@ -177,6 +177,10 @@ export default {
 	justify-content: flex-start;
 	align-items: flex-start;
 
+	h2 {
+		margin-top: 0;
+	}
```
Comment (DorraJaouad) [inline src/components/BreakoutRoomsEditor/BreakoutRoomsEditor.vue:182]: Are you sure it's needed ? it's 0 by default

File: src/components/RightSidebar/BreakoutRooms/BreakoutRoomsActions.vue [hunk 3]
```diff
@@ -291,6 +291,10 @@ export default {
 	&__editor {
 		height: 100%;
 		padding: 20px;
+
+		h2 {
+			margin-top: 0;
+		}
```
Comment (DorraJaouad) [inline src/components/RightSidebar/BreakoutRooms/BreakoutRoomsActions.vue:297]: same here

File: src/stores/breakoutRooms.ts [hunk 4]
```diff
@@ -62,7 +62,8 @@ export const useBreakoutRoomsStore = defineStore('breakoutRooms', {
 
 	getters: {
 		breakoutRooms: (state) => (token: string): BreakoutRoom[] => {
-			return Object.values(Object(state.rooms[token]))
+			const roomsArray: BreakoutRoom[] = Object.values(Object(state.rooms[token]))
+			return roomsArray.sort((roomA, roomB) => roomA.id - roomB.id)
```
Comment (DorraJaouad) [inline src/stores/breakoutRooms.ts:66]: Looks like a good fix to ensure the right order it is expected in modal

**Review by DorraJaouad — COMMENTED** [review]
For better format

```
.participants-editor__buttons {
	padding-top: 10px;

.breakout-rooms-actions__editor {
	padding: 20px 20px 5px;
}

```

File: src/components/BreakoutRoomsEditor/BreakoutRoomsEditor.vue (hunk 2 above)
Comment (Antreesy) [reply src/components/BreakoutRoomsEditor/BreakoutRoomsEditor.vue:182]: I see the default styles from apps.css

image
image

**Review by DorraJaouad — APPROVED** [review]

**Review by DorraJaouad — APPROVED** [review]

**Pull request merged**